chore: Initialize repo using cookiecutter (#3)
openedx/event-bus-kafka: README.rst

> edx_event_bus_kafka

Author: Author note: This should be using `project_title` instead, I think.

=============================

|pypi-badge| |ci-badge| |codecov-badge| |doc-badge| |pyversions-badge|
|license-badge|

The ``README.rst`` file should start with a brief description of the repository,
which sets it in the context of other repositories under the ``edx``
organization. It should make clear where this fits in to the overall edX
codebase.

> Kafka implementation for Open edX event bus.

Author: Author note: I think this should be moved above the placeholder paragraph in the cookiecutter.

Reviewer: Yeah, I'd put it just below the title

Overview (please modify)
------------------------

> The ``README.rst`` file should then provide an overview of the code in this

Reviewer: Note: the README from kafka_consumer is only about the actual consumer, so we need to expand to make it appropriate for this repo.

edX Event Bus - Kafka

> =============================

Author: Style nit: Match length to header text
```suggestion
=====================
```

edx_event_bus_kafka
=============================

|pypi-badge| |ci-badge| |codecov-badge| |doc-badge| |pyversions-badge|
|license-badge|

The ``README.rst`` file should start with a brief description of the repository,
which sets it in the context of other repositories under the ``edx``
organization. It should make clear where this fits in to the overall edX
codebase.

Kafka implementation for Open edX event bus.

Overview (please modify)
------------------------

> This repository details an implementation of the event bus using Kafka.

Reviewer: ```suggestion
This package implements an event bus for Open EdX using Kafka.
```

> Essentially what the event bus does is it processes events from microservices, events can be sent in to the event bus and sent out to be consumed by consumers.

Reviewer: No need for "Essentially". Also I wouldn't use "processes" here, since the event bus doesn't exactly process events, it more queues them up for other services to process.
```suggestion
The event bus acts as a broker between services publishing events and other services that consume these events.
```

> Events can contain modified data which then send signals to be consumed to their respective consumer via the event bus.

Reviewer: I really do not understand this sentence, and I'm not entirely sure it's accurate. Signals send events, not the other way around. Also, event data itself should be immutable, although it may contain information about other data modifications that have occurred.
Basically I think we should scrap this sentence.

Implementing the event bus will allow for asynchronous messaging across services which greatly improves efficiency as we don't have to wait for scheduled batch synchronizations.

> Event bus will have containers that run management commands using Kubernetes by using a loop to listen for new events and run processing code for each event.

Reviewer: ```suggestion
This plugin contains a management command to listen for new events and run processing code for each event using a `while True` loop.
```

> For deployment purposes, the event bus will use SaaS provider confluent.

Reviewer: Remove this sentence. The event bus should work with any Kafka deployment (local or hosted). Remember that ideally this app will be usable by folks outside 2u.

The goal for the Kafka Consumer is to eventually have a flexible event bus that can be easily brought into other apps and repositories to produce and consume arbitrary topics. 
Ideally, the event bus itself will also be an abstraction behind which platform maintainers can use non-Kafka implementations (Redis, Pulsar, etc.). 
The documentation/ADRs may also be moved to more appropriate places as the process matures.

There are a hefty number of "# TODO (EventBus)" annotations left in to help guide further development. 

> This app is intended to be subject to frequent and rapid changes. 

Reviewer: ```suggestion
While still under development, this app will be subject to frequent and rapid changes. 
```

edx_event_bus_kafka
=============================

|pypi-badge| |ci-badge| |codecov-badge| |doc-badge| |pyversions-badge|
|license-badge|

> The ``README.rst`` file should start with a brief description of the repository,

Reviewer: Delete this paragraph

which sets it in the context of other repositories under the ``edx``
organization. It should make clear where this fits in to the overall edX
codebase.

Kafka implementation for Open edX event bus.

Overview (please modify)
------------------------
This package implements an event bus for Open EdX using Kafka.

The event bus acts as a broker between services publishing events and other services that consume these events. 

> Implementing the event bus will allow for asynchronous messaging across services which greatly improves efficiency as we don't have to wait for scheduled batch synchronizations.

Reviewer: We should add something about how this will increase reliability as services will not rely on each other to be available at all times.

> This plugin contains a management command to listen for new events and run processing code (converts signal to message for event bus)

Reviewer: Split into 2:
the publishing code will take events and turn them into messages and send those messages to the broker
the consuming code will poll the broker for messages and turn them into event data that gets emitted in the calling service
We can put the stuff about the while True loop in the consuming code section

> for each event using a `while True` loop. For deployment purposes, the event bus will use SaaS provider confluent.

Reviewer: The actual Kafka host will be configurable. 
No need to mention Confluent.

> The goal for the Kafka Consumer is to eventually have a flexible event bus that can be easily brought into other apps and repositories to produce and consume arbitrary topics. 

Reviewer: the goal for event-bus-kafka

Ideally, the event bus itself will also be an abstraction behind which platform maintainers can use non-Kafka implementations (Redis, Pulsar, etc.). 
The documentation/ADRs may also be moved to more appropriate places as the process matures.

There are a hefty number of "# TODO (EventBus)" annotations left in to help guide further development. 
While still under development, this app will be subject to frequent and rapid changes. 
Outside of testing this app, it is best to leave the KAFKA_CONSUMERS_ENABLED setting off.

The repository works together with the openedx/openedx-events repository to make the fully functional event bus.

Documentation
-------------

> https://open-edx-proposals.readthedocs.io/en/latest/architectural-decisions/oep-0052-arch-event-bus-architecture.html

Reviewer: Just label what this is
